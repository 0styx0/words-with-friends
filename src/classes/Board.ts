import TileInfo from './TileInfo';

export default class Board extends Map {

    constructor(board?: Board) {
        super();

        // copies board
        if (board) {
            board.forEach((value, key) => super.set(key, value));
        }
    }

    set(coordinates: number[], value: TileInfo) {
        return super.set(`${coordinates[0]}, ${coordinates[1]}`, value);
    }

    get(coordinates: number[]): TileInfo | undefined {

        return coordinates ? super.get(`${coordinates[0]}, ${coordinates[1]}`) : new TileInfo();
    }
}