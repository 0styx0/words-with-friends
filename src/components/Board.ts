
class Board {

    dimensions = 15; // square board, so 15x15
}