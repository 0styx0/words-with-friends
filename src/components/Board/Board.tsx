import * as React from 'react';
import './index.css';

import BoardTileHolder from '../TileHolder/TileHolder';

export default function Board() {

    let board: typeof BoardTileHolder[] = [];

    for (let i = 0; i < +process.env.REACT_APP_BOARD_DIMENSIONS!; i++) {

        for (let j = 0; j < +process.env.REACT_APP_BOARD_DIMENSIONS!; j++) {

            board.push(
                <BoardTileHolder
                    coordinates={[j, i]}
                    key={`${j}, ${i}`}
                    {...[] as any}
                /> as {} as typeof BoardTileHolder
            );
        }
    }

    return (
        <div className="wrapper">
            {board}
        </div>
    );
}
