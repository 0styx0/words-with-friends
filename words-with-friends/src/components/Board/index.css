.wrapper {
  display: grid;
  grid-template-columns: repeat(15, 0.5fr);
  grid-gap: 7px;
  grid-auto-rows: minmax(calc(6vh - (15px)), auto);
  max-height: 90vh;
  width: 95%;
  --width-of-player-hand: 13vw; /* width-of-player-hand (defined in Hand/index.css) + 3vw for some space */
  margin: 0 auto;
}

/*.wrapper div {
    border: 1px solid black;
}*/