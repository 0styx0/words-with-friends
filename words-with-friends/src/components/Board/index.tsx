import * as React from 'react';
import './index.css';
import TileHolder from '../TileHolder/TileHolder';
import tilebag from '../../services/tilebag';

export default function Board() {

    tilebag.init(); // move to somewhere else once get game mechanic workings

    // const board = [];

    return (
        <div className="wrapper">
            {new Array(225).fill(null).map((elt, i) => <TileHolder key={i} />)}
        </div>
    );
}
