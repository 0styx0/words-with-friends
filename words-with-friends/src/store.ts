import reducers from './reducers';
import { createStore } from 'redux';

export const defaultState = {
    turn: 0
};

const store = createStore(reducers, defaultState);

export default store;
