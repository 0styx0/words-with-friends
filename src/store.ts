import reducers from './reducers';
import { createStore, compose } from 'redux';
import Player from './classes/Player';
import Tilebag from './classes/Tilebag';
import Board from './classes/Board';

const enhancers = compose('devToolsExtension' in window ?
    (window as {} as { devToolsExtension: Function }).devToolsExtension!() :
    (f: {}) => f);

export const defaultState = {
    turn: 0,
    Players: [new Player(true, 0), new Player(false, 1)], // turns switched around on INIT_PLAYERS
    board: new Board(),
    Tilebag: new Tilebag()
};

const store = createStore(reducers, defaultState, enhancers);

export default store;
