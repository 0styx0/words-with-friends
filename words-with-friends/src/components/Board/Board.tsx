import * as React from 'react';
import Board from './';
import TileHolder from '../TileHolder/TileHolder';
import Game from '../../classes/Game';

interface State {
    board: typeof TileHolder[];
}

export default class BoardContainer extends React.Component<{}, State> {

    constructor() {
        super();

        this.state = {
            board: []
        };
    }

    componentDidMount() {

        let board: typeof TileHolder[] = [];

        for (let i = 0; i < 15; i++) {

            board.push([] as any); // board = [[]]

            for (let j = 0; j < 15; j++) {

                board[i][j] = <TileHolder coordinates={`${i}, ${j}`} canDrag={true} key={`${i}, ${j}`} />;

                // const centerTile = i === 7 && j === 7;

                Game.board.set(`${i}, ${j}`, {
                    filled: false,
                    turnTileWasPlaced: 0,
                    recent: false
                });
            }
        }

        this.setState({
            board
        });
    }

    render() {
        return <Board tiles={this.state.board} />;
    }
}