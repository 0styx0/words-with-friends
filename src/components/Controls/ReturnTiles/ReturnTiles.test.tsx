import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store, { getState } from '../../../store';
import { mount } from 'enzyme';
import ConnectedReturnTiles from './ReturnTiles';
import placeWord from '../../../test/helpers/placeWord';
import getWord from '../../../test/helpers/getWord';
import addCoordinatesToStore from '../../../test/helpers/addCoordinatesToStore';
import Board from '../../../classes/Board';

describe(`<ReturnTiles />`, () => {

    describe('render', () => {

        it('renders properly', () => {

            const tree = renderer.create(
                <Provider store={store} >
                  <ConnectedReturnTiles {...[] as any} />
                </Provider>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });

    function setup() {

        return mount(
            <Provider store={store} >
              <ConnectedReturnTiles {...[] as any} />
            </Provider>
        );
    }

    describe(`#returnTiles`, () => {

        const wrapper = setup();
        const centerCoordinates =
          [+process.env.REACT_APP_CENTER_COORDINATE!, +process.env.REACT_APP_CENTER_COORDINATE!];

        it(`it returns tiles placed in last turn to current player's hand`, () => {

            const { tileInfos, coordinates } = placeWord(getWord(), centerCoordinates, true, new Board(), 1);
            addCoordinatesToStore(tileInfos, coordinates);

            const playerTilesBeforeTileRecall = getState().Players[0].tiles;
            expect(playerTilesBeforeTileRecall).toHaveLength(0);

            wrapper.find('button').simulate('click');

            const playerTilesAfterTileRecall = getState().Players[0].tiles;
            expect(playerTilesAfterTileRecall).toEqual(tileInfos.map(tileInfo => tileInfo.tile));
        });


        it(`takes those tiles off the board`, () => {

            const { tileInfos, coordinates } = placeWord(getWord(), centerCoordinates, true, new Board(), 1);
            addCoordinatesToStore(tileInfos, coordinates);

            wrapper.find('button').simulate('click');

            coordinates.forEach(coordinate => {
                expect(getState().board.get(coordinate)!.filled).toBeFalsy();
            });

        });
    });
});
