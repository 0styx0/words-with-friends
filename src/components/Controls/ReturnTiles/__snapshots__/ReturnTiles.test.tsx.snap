// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReturnTiles /> render renders properly 1`] = `
<button
  onClick={[Function]}
>
  Return Tiles
</button>
`;
